File: iosApp/iosApp/Features/Profile/components/LevelBlock.swift
//
//  LevelBlock.swift
//  iosApp
//
//  Created by velkonost on 09.11.2023.
//  Copyright © 2023 orgName. All rights reserved.
//

import Foundation
import SwiftUI
import SharedSDK

struct LevelBlock: View {
    
    private let experienceData: ExperienceData
    private let topPadding: CGFloat
    private let isOwn: Bool
    
    init(experienceData: ExperienceData, topPadding: Int = 20, isOwn: Bool = true) {
        self.experienceData = experienceData
        self.topPadding = CGFloat(topPadding)
        self.isOwn = isOwn
    }
    
    var body: some View {
        
        PrimaryBox {
            VStack(spacing: 0) {
                HStack {
                    Text(
                        isOwn ? SharedR.strings().experience_your_title.desc().localized() : SharedR.strings().experience_user_title.desc().localized()
                    )
                        .style(.labelMedium)
                        .foregroundColor(.textPrimary)
                    Spacer()
                    Text(experienceData.currentLevelStr.localized())
                        .style(.labelMedium)
                        .foregroundColor(.textPrimary)
                }
                
                ProgressView(value: Double(experienceData.remainExperiencePercent))
                    .padding(.top, 6)
                    .accentColor(.buttonGradientStart)
                    
            }
        }
        .padding(.top, topPadding)
    }
}

File: androidApp/core/compose/src/main/kotlin/com/velkonost/getbetter/core/compose/components/experience/LevelBlock.kt
package com.velkonost.getbetter.core.compose.components.experience

import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.material3.LinearProgressIndicator
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.remember
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.StrokeCap
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.unit.Dp
import com.velkonost.getbetter.core.compose.components.PrimaryBox
import com.velkonost.getbetter.core.compose.components.WeightedSpacer
import com.velkonost.getbetter.core.compose.theme.Dimen.DP_20
import com.velkonost.getbetter.core.compose.theme.Dimen.DP_8
import com.velkonost.getbetter.shared.core.model.user.ExperienceData
import com.velkonost.getbetter.shared.resources.SharedR
import dev.icerock.moko.resources.compose.colorResource
import dev.icerock.moko.resources.compose.stringResource

@Composable
fun LevelBlock(
    modifier: Modifier = Modifier,
    topPadding: Dp = DP_20,
    experienceData: ExperienceData,
    isOwn: Boolean = true
) {

    val indicatorTopPadding = remember { DP_8 }
    val context = LocalContext.current

    PrimaryBox(modifier = modifier.padding(top = topPadding)) {
        Column {
            Row {
                Text(
                    text = stringResource(
                        resource =
                        if (isOwn) SharedR.strings.experience_your_title
                        else SharedR.strings.experience_user_title
                    ),
                    style = MaterialTheme.typography.labelMedium,
                    color = colorResource(resource = SharedR.colors.text_primary)
                )
                WeightedSpacer()
                Text(
                    text = experienceData.currentLevelStr.toString(context),
                    style = MaterialTheme.typography.labelMedium,
                    color = colorResource(resource = SharedR.colors.text_primary)
                )
            }

            LinearProgressIndicator(
                modifier = modifier
                    .fillMaxWidth()
                    .padding(top = indicatorTopPadding),
                progress = experienceData.remainExperiencePercent,
                color = colorResource(resource = SharedR.colors.button_gradient_start),
                trackColor = colorResource(resource = SharedR.colors.text_unimportant_color),
                strokeCap = StrokeCap.Round
            )
        }
    }

}